### PR TITLE
Improvements to dat downloading and historic-cache management

### PR DESCRIPTION
**Pull request by pfrazee**
Currently, Beaker handles syncing a 'saved' Dat by using the staging area's diff and revert(). We discovered the diff handles large dats very poorly. The solution also caused all dat files to be duplicated, which was subpar. In this PR, we add a faster algorithm for syncing latest, and also improve the deletion of historic files in a Dat.

Still TODO

 - [ ] Unlink from latest dir
 - [ ] A way to change the storage configuration when a Dat's localPath changes
 - [ ] Update the garbage collector to remove historic files
 - [ ] Automatically download latest, perhaps by using hyperdrive `latest=true` or by copying key parts of that algorithm
 - [ ] Turn our storage code into a module
 - [ ] Update multi-random-access to tell whether a read or write is being done, and then only replace hard links if a write is occurring

### CODE REVIEW

File: app/background-process/networks/dat/storage.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        if (err) return cb(err)
+        if (!node) return cb(new Error('Could not locate data'))
+
+        // provide the storage
+        var historicFilePath = path.join(metadataDir, 'content', 'files', `${node.name}.${st.offset}`)
+        cb(null, {
+          start: st.byteOffset,
+          end: st.byteOffset + st.size,
+          storage: raf(historicFilePath)
+        })
```
Comment (mafintosh) [inline app/background-process/networks/dat/storage.js]: my gut tells me we should wait on calling the callback until after the hardlinker has run ... for safety reasons

Comment (pfrazee) [inline app/background-process/networks/dat/storage.js]: You think? No code should rely on those hardlinks: they exist purely for the end-user to be able to access the files in a convenient place

Comment (mafintosh) [inline app/background-process/networks/dat/storage.js]: I think this might bite us in the future when we start doing deletes otherwise

**Comment by pfrazee** [comment]
We gave this some thought and decided the core UX was off. The sites directory is a bit confusing.

We're instead going to aim for an experience similar to what's suggested in https://github.com/beakerbrowser/beaker/issues/508#issuecomment-307128116, and then add controls to download sites as a zip.

**Pull request closed without merging**